index.js

```diff
@@ -1,59 +1,108 @@
 const Executor = require('screwdriver-executor-base');
+const Joi = require('joi');
+const Resque = require('node-resque');
+
+/**
+ * Strip the leading 'screwdriver-executor' from plugin names
+ * @method getPluginName
+ * @param {String} plugin
+ * @return {String}
+ */
+function getPluginName(plugin) {
+    return plugin.replace('screwdriver-executor-', '');
+}
 
-class ExecutorRouter extends Executor {
+class ExecutorQueue extends Executor {
     /**
      * Constructs a router for different Executor strategies.
      * @method constructor
      * @param  {Object}         config                      Object with executor and ecosystem
      * @param  {String}         [config.defaultPlugin]      Optional default executor
-     * @param  {Object}         [config.ecosystem]          Optional object with ecosystem values
-     * @param  {Array|Object}   config.executor             Array of executors to load or a single executor object
+     * @param  {Object}         config.ecosystem            Object with ecosystem values
+     * @param  {Array}          config.executor             Array of executors to load
+     * @param  {Object}         config.redisConnection      Connection details for redis
      * @param  {String}         config.executor[x].name     Name of the executor NPM module to load
      * @param  {String}         config.executor[x].options  Configuration to construct the module with
      */
     constructor(config = {}) {
-        const ecosystem = config.ecosystem || {};
         const executorConfig = config.executor;
-        const defaultPlugin = config.defaultPlugin;
+        const ecosystem = config.ecosystem || {};
 
         if (!executorConfig || !(Array.isArray(executorConfig)) || executorConfig.length === 0) {
             throw new Error('No executor config passed in.');
         }
 
         super();
 
-        executorConfig.forEach((plugin) => {
-            let ExecutorPlugin;
+        this.ecosystem = ecosystem;
+        this.defaultPlugin = config.defaultPlugin;
 
-            try {
-                // eslint-disable-next-line global-require, import/no-dynamic-require
-                ExecutorPlugin = require(`screwdriver-executor-${plugin.name}`);
-            } catch (err) {
-                console.error(err);
+        this.executorConfigs = {};
+        executorConfig.forEach((plugin) => {
+            this.executorConfigs[plugin.name] = Object.assign({ ecosystem }, plugin.options);
 
-                return;
+            if (!this.defaultPlugin) {
+                this.defaultPlugin = plugin.name;
             }
+        });
+
+        // eslint-disable-next-line new-cap
+        this.queue = new Resque.queue({ connection: config.redisConnection });
 
-            const options = Object.assign({ ecosystem }, plugin.options); // Add ecosystem to executor options
-            const executorPlugin = new ExecutorPlugin(options);
+        this.connected = false;
+        this.queue.connect(() => {
```

Comment:
I don't think there's much we can do about it in the constructor (since we don't want the constructor to be async). If the connection fails it will try to connect once somebody tries to enqueue something.